engine/src/main/java/org/terasology/rendering/dag/nodes/ApplyDeferredLightingNode.java

```diff
@@ -45,46 +46,53 @@
+        refractiveReflectiveFbo = displayResolutionDependentFBOs.get(REFRACTIVE_REFLECTIVE);
 
-        addDesiredStateChange(new SetViewportToSizeOf(WRITE_ONLY_GBUFFER));
+        addDesiredStateChange(new SetViewportToSizeOf(SCENE_OPAQUE_PING_PONG, displayResolutionDependentFBOs));
         addDesiredStateChange(new EnableMaterial(DEFERRED_LIGHTING_MATERIAL.toString()));
 
         int textureSlot = 0;
         addDesiredStateChange(new SetInputTexture(
-                textureSlot++, READ_ONLY_GBUFFER.getFbo().colorBufferTextureId,   DEFERRED_LIGHTING_MATERIAL, "texSceneOpaque"));
+                textureSlot++, sceneOpaqueFbo.colorBufferTextureId,   DEFERRED_LIGHTING_MATERIAL, "texSceneOpaque"));
         addDesiredStateChange(new SetInputTexture(
-                textureSlot++, READ_ONLY_GBUFFER.getFbo().depthStencilTextureId,  DEFERRED_LIGHTING_MATERIAL, "texSceneOpaqueDepth"));
+                textureSlot++, sceneOpaqueFbo.depthStencilTextureId,  DEFERRED_LIGHTING_MATERIAL, "texSceneOpaqueDepth"));
         addDesiredStateChange(new SetInputTexture(
-                textureSlot++, READ_ONLY_GBUFFER.getFbo().normalsBufferTextureId, DEFERRED_LIGHTING_MATERIAL, "texSceneOpaqueNormals"));
+                textureSlot++, sceneOpaqueFbo.normalsBufferTextureId, DEFERRED_LIGHTING_MATERIAL, "texSceneOpaqueNormals"));
         addDesiredStateChange(new SetInputTexture(
-                textureSlot,   READ_ONLY_GBUFFER.getFbo().lightBufferTextureId,   DEFERRED_LIGHTING_MATERIAL, "texSceneOpaqueLightBuffer"));
+                textureSlot,   sceneOpaqueFbo.lightBufferTextureId,   DEFERRED_LIGHTING_MATERIAL, "texSceneOpaqueLightBuffer"));
     }
 
     /**
      * Part of the deferred lighting technique, this method applies lighting through screen-space
-     * calculations to the previously flat-lit world rendering, stored in the READ_ONLY_GBUFFER.
+     * calculations to the previously flat-lit world rendering, stored in the engine:sceneOpaque.
      * <p>
      * See http://en.wikipedia.org/wiki/Deferred_shading for more information on the general subject.
      */
     @Override
     public void process() {
         PerformanceMonitor.startActivity("rendering/applyDeferredLighting");
 
-        WRITE_ONLY_GBUFFER.bind(); // TODO: remove and replace with a state change
-        WRITE_ONLY_GBUFFER.setRenderBufferMask(true, true, true);
+        sceneOpaquePingPongFbo.bind(); // TODO: remove and replace with a state change
+        sceneOpaquePingPongFbo.setRenderBufferMask(true, true, true);
         glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT); // TODO: this is necessary - but why? Verify in the shader.
 
         renderFullscreenQuad();
 
-        displayResolutionDependentFBOs.swapReadWriteBuffers();
-        READ_ONLY_GBUFFER.attachDepthBufferTo(displayResolutionDependentFBOs.get(REFRACTIVE_REFLECTIVE));
+        displayResolutionDependentFBOs.swapReadWriteBuffers(); // Doesn't exactly help with a lean process() function..
```

Comment:
Ah, this was meant as a comment for you. NOT meant to be a part of the final code. My bad 😓